stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -1172,6 +1172,65 @@ public func _TFCGetCTensorHandleFromSwift(
   return handle.cTensorHandle
 }
 
+/// Adds `handle` as an input to `op`.
+@usableFromInline
+@_silgen_name("_swift_tfc_OpAddInputFromTensorHandle")
+func _TFCOpAddInputFromTensorHandle(_ op: CTFEOp,
+                                    _ handle: _AnyTensorHandle,
+                                    _ status: CTFStatus) {
+  TFE_OpAddInput(op, handle.cTensorHandle, status)
+}
+
+/// Adds `t` as an input or inputs to `op`.
+@usableFromInline
+@_silgen_name("_swift_tfc_OpAddInputFromTensorGroup")
+func _TFCOpAddInputFromTensorGroup<T: TensorGroup>(
+    _ op: CTFEOp, _ t: T, _ status: CTFStatus
+) -> Int32 {
+  let count = T._tensorHandleCount
+  let buffer = UnsafeMutablePointer<CTensorHandle>.allocate(capacity: Int(count))
+  defer { buffer.deallocate() }
+  t._unpackTensorHandles(resultBuffer: buffer)
+  for i in 0..<count {
+    TFE_OpAddInput(op, buffer.advanced(by: Int(i)).pointee, status)
+    if TF_GetCode(status) != TF_OK {
+      return 0
+    }
+  }
+  return count
+}
+
+/// Initializes a TensorGroup value, taking ownership of all the tensor
+/// handles in `tensorHandles`. Also deallocates `tensorHandles`.
+@usableFromInline
+@_silgen_name("_swift_tfc_InitTensorGroup")
+func _TFCInitTensorGroup<T: TensorGroup>(
+    owning tensorHandles: UnsafeMutablePointer<CTensorHandle>
```

Comment:
I'd suggest dropping argument labels for top-level, runtime entry functions, because they are no direct callers from Swift.